File: kameleon_PLUS_interpolator/Kameleon_Pro/src/ccmc/Attribute.cpp
#include "Attribute.h"
#include <iostream>

/**
 *
 */
Attribute::Attribute()
{
	attributeName = "";
	sValue = "";
	iValue = 0;
	fValue = 0.f;

}

/**
 * @return
 */
std::string Attribute::getAttributeName()
{
	return attributeName;
}

/**
 * @param name The attribute name requested.
 */
void Attribute::setAttributeName(std::string name)
{
	attributeName = name;
}

/**
 * Copies the contents of value and stores them.
 * @param value the new attribute value
 */
void Attribute::setAttributeValue(std::string& value)
{
	type = Attribute::STRING;
	sValue = value;
}

/**
 * Copies the contents of value and stores them.
 * @param value the new attribute value
 */
void Attribute::setAttributeValue(int& value)
{
	type = Attribute::INT;
	iValue = value;
}

/**
 * Copies the contents of value and stores them.
 * @param value the new attribute value
 */
void Attribute::setAttributeValue(float& value)
{
	type = Attribute::FLOAT;
	fValue = value;
}


/**
 * @return AttributeType of the Attribute object
 */
Attribute::AttributeType Attribute::getAttributeType()
{
	return type;
}

/**
 * Returns the attribute value as a float, if applicable.
 * @return The float value of the attribute.  This value returned will be 0.f
 * if the AttributeType of the Attribute object is not Attribute::FLOAT
 */
float Attribute::getAttributeFloat()
{
	return fValue;
}

/**
 * @return
 */
std::string Attribute::getAttributeString()
{
	return sValue;
}

/**
 * @return
 */
int Attribute::getAttributeInt()
{
	return iValue;
}

/**
 * Destructor
 */
Attribute::~Attribute()
{
	/*
	if (attributeValue != NULL)
	{
		std::cout << "deleting attributeValue in Attribute.cpp" << std::endl;
		if (type == Attribute::STRING)
		{
			delete [] (char *)attributeValue;
			attributeValue = NULL;
		}else if(type == Attribute::INT)
		{
			delete (int *)attributeValue;
			attributeValue = NULL;
		}
		else
		{
			delete (float *)attributeValue;
			attributeValue = NULL;
		}
	}*/
}

File: kameleon-plus/trunk/kameleon-plus-working/src/ccmc/pyreaders/Attribute.py
import unittest
import sys, os
sys.path.append('build')
sys.path.append('.')
sys.path.append('../')
import pyKameleon


class Attribute(pyKameleon.Attribute):
	"""
	python Attribute class accessible from C++

	AttributeType       	attributeName       	getAttributeName    	           		        setAttrStr
	FLOAT               	fValue              	getAttributeString  	sValue              	setAttributeName
	INT                 	getAttributeFloat   	getAttributeType    	setAttrFloat        	toString
	STRING              	getAttributeInt     	iValue              	setAttrInt          	type

	"""
	def __init__(self, name = None, value = None):
		"""default Attribute constructor
		"""
		super(Attribute, self).__init__()
		if name!= None:
			self.setAttributeName(name)
		if type(value) == int:
			self.setAttrInt(value)
		elif type(value) == float:
			self.setAttrFloat(value)
		elif type(value) == str:
			self.setAttrStr(value)
		elif type(value) == unicode:
			self.setAttrStr(str(value))

	def getAttributeValue(self):
		if self.getAttributeType() == pyKameleon.Attribute.STRING:
			return self.getAttributeString()
		elif self.getAttributeType() == pyKameleon.Attribute.FLOAT:
			return self.getAttributeFloat()
		elif self.getAttributeType() == pyKameleon.Attribute.INT:
			return self.getAttributeInt()


class Test_pyReader(unittest.TestCase):
	def setUp(self):
		self.fruit_attr = Attribute('fruit','apple')
		self.size_attr = Attribute('radius', 3.0)
		self.int_attr = Attribute('count', 5)

	def test_attribute_name(self):
		print '\tchecking that attribute name matches'
		self.assertEqual(self.fruit_attr.getAttributeName(), 'fruit')

	def test_attribute_str(self):
		print '\tchecking that string value matches'
		self.assertEqual(self.fruit_attr.getAttributeType(), pyKameleon.Attribute.STRING)
		self.assertEqual(self.fruit_attr.getAttributeString(), 'apple')

	def test_attribute_float(self):
		print '\tchecking that float value matches'
		self.assertEqual(self.size_attr.getAttributeType(), pyKameleon.Attribute.FLOAT)
		self.assertEqual(self.size_attr.getAttributeFloat(), 3.0)

	def test_attribute_int(self):
		print '\tchecking that int value matches', self.int_attr.iValue
		self.assertEqual(self.int_attr.getAttributeType(), pyKameleon.Attribute.INT)
		self.assertEqual(self.int_attr.getAttributeInt(), 5)

	def test_getAttributeValue(self):
		print '\tchecking that arbitrary value can be returned'
		self.assertEqual(self.fruit_attr.getAttributeValue(), 'apple')
		self.assertEqual(self.size_attr.getAttributeValue(), 3.0)
		self.assertEqual(self.int_attr.getAttributeValue(), 5)

	def test_attribute_is_subclass(self):
		print '\tchecking that attribute is a subclass of pyKameleon.Attribute'
		self.assertTrue(issubclass(self.fruit_attr.__class__, pyKameleon.Attribute))


def main():
	print "Performing tests for Attribute"
	unittest.main()


if __name__ == '__main__':
	main()
